### PR TITLE
daemon: factor out node config headerfile into separate function

### PR DESCRIPTION
**Pull request by ianvernon**
This makes the code more readable / modular.

Signed-off by: Ian Vernon <ian@cilium.io>

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Review by raybejjani — APPROVED** [review]

**Review by aanm — COMMENTED** [review]
Cool! `createNodeConfigHeaderfile` only needs the `d.loopbackIPv4` can we put the `d.lookbackIPv4` in the `pkg/node` and move this whole function to elsewhere like `pkg/bpf`?

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
@aanm 

> createNodeConfigHeaderfile only needs the d.loopbackIPv4 can we put the d.lookbackIPv4 in the pkg/node

Done.

>  and move this whole function to elsewhere like pkg/bpf?

I don't want to move this to `pkg/bpf` because that will introduce `pkg/node` and its dependencies as dependencies into the BPF package, which I am not a fan of. I'm keeping this as is for now.

File: daemon/daemon.go [hunk 1]
```diff
@@ -1145,8 +1157,8 @@ func NewDaemon() (*Daemon, error) {
 		if err != nil {
 			return nil, fmt.Errorf("Unable to reserve IPv4 loopback address: %s", err)
 		}
-		d.loopbackIPv4 = loopbackIPv4
-		log.Infof("  Loopback IPv4: %s", d.loopbackIPv4.String())
+		node.SetIPv4Loopback(loopbackIPv4)
```
Comment (aanm) [inline daemon/daemon.go:1159]: `d.loopbackIPv4` can be removed now

**Review by aanm — DISMISSED** [review]

**Comment by ianvernon** [comment]
test-me-please

File: daemon/daemon.go (hunk 1 above)
Comment (ianvernon) [reply daemon/daemon.go:1159]: Done.

**Review by aanm — APPROVED** [review]

**Review by eloycoto — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Pull request merged**